campari/io.py

```
@@ -116,30 +148,35 @@ def build_lightcurve_sim(supernova, flux, sigma_flux):
     return QTable(data=data_dict, meta=meta_dict, units=units)
 
 
-def save_lightcurve(lc=None, identifier=None, psftype=None, output_path=None, overwrite=True):
+def save_lightcurve(lc=None, identifier=None, psftype=None, output_path=None, overwrite=True, save_to_database=False):
     """This function parses settings in the SMP algorithm and saves the
     lightcurve to an ecsv file with an appropriate name.
     Input:
-    lc: the lightcurve data
+    lc: the lightcurve data, in the form of a snappl.lightcurve.Lightcurve object
     identifier (str): the supernova ID or "simulated"
     band (str): the bandpass of the images used
     psftype (str): "romanpsf" or "analyticpsf"
     output_path (str): the path to save the lightcurve to.  Defaults to
-      config value phtometry.campari.paths.output_dir
+      config value system.paths.lightcurves
 
     Returns:
     None, saves the lightcurve to a ecsv file.
     The file name is:
     <output_path>/identifier_band_psftype_lc.ecsv
     """
-    band = lc["filter"][0]
-    output_path = Config.get().value("photometry.campari.paths.output_dir") if output_path is None else output_path
-    output_path = pathlib.Path(output_path)
-    output_path.mkdir(exist_ok=True, parents=True)
-
-    lc_file = output_path / f"{identifier}_{band}_{psftype}_lc.ecsv"
-    SNLogger.info(f"Saving lightcurve to {lc_file}")
-    lc.write(lc_file, format="ascii.ecsv", overwrite=overwrite)
+    band = lc.meta["band"]
+    SNLogger.debug(f"saving lightcurve for id={identifier}, band={band}, psftype={psftype}")
+    base_output_path = Config.get().value("system.paths.lightcurves") if output_path is None else output_path
```

As above : you should only use `system.paths.lightcurves` if `save_to_database` is True.

(In fact, also, if save_to_database is true and `output_path` is not `None`, then you should raise an exception.)

---

done!